time_series_anomaly_detection/detector.py

```diff
@@ -1,42 +1,731 @@
+    Reshape
+from tensorflow.keras.optimizers import Adam, RMSprop
+from sklearn.preprocessing import MinMaxScaler, StandardScaler
+import matplotlib.pyplot as plt
+from datetime import date
+from typing import Dict
 
 from time_series_anomaly_detection.abstractions import (
     TimeSeriesAnomalyDetector
 )
 
 
-class TemplateDetector(TimeSeriesAnomalyDetector):
+class GAN_AD(TimeSeriesAnomalyDetector):
     """
-    Time series anomaly detector template.
-
-    Parameters
-    ----------
-    id_columns: Iterable[str], optional
-        ID columns used to identify individual time series.
-
-        Should be specified in case the detector is provided with
-        time series during training or inference with ID columns
-        included. Using these columns the detector can separate individual
-        time series and not use ID columns as feature columns.
-        In case they are not specified, all columns are regarded as feature
-        columns and the provided data is regarded as a single time series.
+    Time series GAN anomaly detector.
     """
 
     def __init__(
-        self,
-        id_columns: Optional[Iterable[str]] = None,
+            self,
+            window_size: int = 8,
+            shift: int = 3,
+            batch_size: int = 8,
+            latent_dim: int = 1,
+            discriminator_params: Dict = {'lstm_units': 150,
+                                          'dense_units': 50,
+                                          'dropout_rate': 0.2},
+            generator_params: Dict = {'conv_filters': 30,
+                                      'conv_kernel_size': 3,
+                                      'lstm_units': 200},
+            id_columns: Optional[Iterable[str]] = None
     ):
+        """
+        Inits GAN_AD class with parameters which are related to
+        the training process and data supplied later.
+        Parameters
+        ----------
+        window_size : int, default 8
+            Length of the window taken across time series sequences.
+            The window is applied to both train and test data in order
+            to subdivide long sequences into smaller time series.
+        shift : int, default 3
+            Length of window shift applied to both train in order
+            to subdivide long sequences into smaller time series.
+        batch_size : int, default 8
+            The number of time series samples in training dataset batch.
+        latent_dim : int, default 1
+            Latent space dimension used during GAN-AD model training.
+        discriminator_params : Dict[str, int], default {'lstm_units': 150, 
+                               'dense_units': 50, 'dropout_rate': 0.2}
+            Discriminator hyperparameters specification.                    
+        generator_params : Dict[str, int], default {'conv_filters': 30,
+                           'conv_kernel_size': 3, 'lstm_units': 200}
+            Generator hyperparameters specification.                
```

Comment:
It is not clear from this what parameters are there to set (except for the default ones). Briefly describe the generator / discriminator architectures and all its configurable parameters. 

Also type hint is invalid as _dropout_rate_ is not int (Dict[str, Any] or Dict[str, Union[int, float]] might work).